tests/h/search/query_test.py

```diff
@@ -89,12 +89,25 @@ def search(self, search, pyramid_request):
 
 
 class TestGroupFilter(object):
+    def test_matches_only_annotations_from_specified_group(self, search, Annotation, group):
+        Annotation(groupid='group2')
+        Annotation(groupid='group3')
+        group1_annotations = [Annotation(groupid=group.name).id,
+                              Annotation(groupid=group.name).id]
+
+        result = search.run({'group': group.pubid})
+
+        assert set(result.annotation_ids) == set(group1_annotations)
 
     @pytest.fixture
     def search(self, search):
         search.append_filter(query.GroupFilter())
         return search
 
+    @pytest.fixture
+    def group(self, factories):
+        return factories.OpenGroup(name="group1", pubid="group1")
```

Comment:
The name (the user-visible name) and pubid should be different, as this will reveal an error in the `test_matches_only_annotations_from_specified_group` test above.